Tella/Data/Networking/Resources/ResourceService.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  ResourceService.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/16/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class ResourceService {
+    private var cancellables: Set<AnyCancellable> = []
+    func getDownloadedResources(from appModel: MainAppModel) -> [DownloadedResourceCardViewModel] {
+        guard let resources = appModel.vaultManager.tellaData?.getResources() else {
```

Comment:
I suggest this code 
    
```
 func getDownloadedResources(from appModel: MainAppModel) -> [DownloadedResourceCardViewModel] {
        let resources = appModel.vaultManager.tellaData?.getResources()
        let cardsViewModel = resources?.compactMap { resource in
            let selectedServer = appModel.vaultManager.tellaData?.tellaServers.value.first { $0.id == resource.serverId }
            return DownloadedResourceCardViewModel(resource: resource, server: selectedServer)
        }
        return cardsViewModel ?? []
    }
```

and after inscluding the `server` to `DownloadedResource` it will be 

```
 func getDownloadedResources(from appModel: MainAppModel) -> [DownloadedResourceCardViewModel] {
        let resources = appModel.vaultManager.tellaData?.getResources()
        let cardsViewModel = resources?.compactMap { resource in
            return DownloadedResourceCardViewModel(resource: resource)
        }
        return cardsViewModel ?? []
    }

```

Reply:
I think it's better to init the DownloadedResourceCardViewModel with resource (and if there are other attributes) to get a simple  `getDownloadedResources` function
and I suggetc to delete the  non used attributes like `size`, `createdAt` 

I suggest `ResourceCardViewModel` that contains all the attributes and will be used in the downloaded and availble resources:

```
class ResourceCardViewModel: Hashable, Identifiable {
    var id: String
    var externalId: String
    var fileName: String
    var title: String
    var serverName: String
    var type: ResourceCardType
    var action: () -> Void
    
    init(resource: Resource, action:() -> Void) {
        self.id = resource.id
        self.externalId = resource.externalId
        self.fileName = resource.fileName
        ...
    }
}
```

Reply:
so, in this case we assume that external id is present also in the resource class. But currently is not, because external id exists only in our database. Not in the tella server. 

Should we add the external id also in the resource class and add set it as nil for the available resources? Or put the same value as the id?